xml/issue2158.xml

```diff
@@ -47,9 +47,113 @@ Maybe it would be useful to introduce a new name for such conditional requiremen
 "<tt>CopyInsertable</tt>" and "<tt>MoveInsertable</tt>").
 </p>
 
+
+<note>2016-08 Chicago</note>
+<p>
+The problem does not appear to be as severe as described.  The MoveInsertable
+requirements are consistently correct, but an issue may arise on the
+exception-safety guarantees when we check for
+<tt>is_copy_constructible_v&lt;T&gt;</tt>.  The problem, as described, is
+typically for templates that appear to have a copy constructor, but one that
+fails to compile once instantiated, and so gives a misleading result for the
+trait.
+</p>
+<p>
+In general, users should not provide such types, and the standard would not
+serve users well by trying to address support for such types.  However, the
+standard should not be providing such types either, such as
+<tt>vector&lt;unique_ptr&lt;T&gt;&gt;</tt>.  A possible resolution would be
+to tighten the constraints in Table 80 - Container Requirements, so that if
+the Requirements for the copy constructor/assingment operator of a container
+are not satisfied, that operation shall be deleted.
+</p>
+<p>
+A futher problem highlighted by this approach is that there are no constraints on
+the copy-assignment operator, so that <tt>vector&lt;unique_ptr&lt;T&gt;&gt;</tt>
+should be CopyAssignable!  However, we can lift the equivalent constraints from
+the Allocator-aware container requirements.
+</p>
+
 </discussion>
 
 <resolution>
+
+<p>
+<table border="1">
+<caption> <sref ref="[container.requirements.general]"/> Table 80 &mdash; Container requirements </caption>
+<tr>
+  <td><b>Expression</b></td>
+  <td><b>Return type</b></td>
+  <td><b>Operational semantics</b></td>
+  <td><b>Assertion/note/pre-/post-condition</b></td>
+  <td><b>Complexity</b></td>
+</tr>
+<tr>
+  <td>X(a)</td>
+  <td></td>
+  <td></td>
+  <td>
+    <i>Requires:</i> <tt>T</tt> is <tt>CopyInsertable</tt> into
+    <tt>X</tt> (see below)<del>.</del><ins>, otherwise this expression shall be ill-formed.</ins><br/>
+    post: <tt>a == X(a)</tt>.
+  </td>
+  <td>linear</td>
+</tr>
+<tr>
+  <td><tt>X u(a)</tt><br/><tt>X u = a;</tt></td>
+  <td></td>
+  <td></td>
+  <td>
+    <i>Requires:</i> <tt>T</tt> is <tt>CopyInsertable</tt> into
+    <tt>X</tt> (see below)<del>.</del><ins>, otherwise this expression shall be ill-formed.</ins><br/>
```

Comment:
`CopyInsertable` has a semantic component that can't be checked, and even `allocator_traits<A>::construct(m, p, v)` isn't checkable in the general case.